app/src/main/java/org/thoughtcrime/securesms/util/DateUtils.kt

```diff
@@ -0,0 +1,173 @@
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+
+// Enums used to get the locale-aware String for one of the three relative days
+enum class RelativeDay { TODAY, YESTERDAY, TOMORROW }
+
+/**
+ * Utility methods to help display dates in a nice, easily readable way.
+ */
+object DateUtils : android.text.format.DateUtils() {
+
+    @Suppress("unused")
+    private val TAG: String = DateUtils::class.java.simpleName
+    private val DAY_PRECISION_DATE_FORMAT = SimpleDateFormat("yyyyMMdd")
+    private val HOUR_PRECISION_DATE_FORMAT = SimpleDateFormat("yyyyMMddHH")
+
+    private fun isWithin(millis: Long, span: Long, unit: TimeUnit): Boolean {
+        return System.currentTimeMillis() - millis <= unit.toMillis(span)
+    }
+
+    private fun isYesterday(`when`: Long): Boolean {
+        return isToday(`when` + TimeUnit.DAYS.toMillis(1))
+    }
+
+    private fun convertDelta(millis: Long, to: TimeUnit): Int {
+        return to.convert(System.currentTimeMillis() - millis, TimeUnit.MILLISECONDS).toInt()
+    }
+
+    // Method to get the String for a relative day in a locale-aware fashion
+    public fun getLocalisedRelativeDayString(relativeDay: RelativeDay): String {
+
+        val now = Calendar.getInstance()
+
+        // To compare a time to 'now' we need to use get a date relative it, so plus or minus a day, or not
+        val dayAddition = when (relativeDay) {
+            RelativeDay.TOMORROW  -> {  1 }
+            RelativeDay.YESTERDAY -> { -1 }
+            else -> 0 // Today
+        }
+
+        val comparisonTime = Calendar.getInstance().apply {
+            add(Calendar.DAY_OF_YEAR, dayAddition)
+            set(Calendar.HOUR_OF_DAY, 0)
+            set(Calendar.MINUTE, 0)
+            set(Calendar.SECOND, 0)
+            set(Calendar.MILLISECOND, 0)
+        }
+
+        return getRelativeTimeSpanString(comparisonTime.timeInMillis,
+            now.timeInMillis,
+            DAY_IN_MILLIS,
+            FORMAT_SHOW_DATE).toString()
+    }
+
+    fun getFormattedDateTime(time: Long, template: String, locale: Locale): String {
+        val localizedPattern = getLocalizedPattern(template, locale)
+        return SimpleDateFormat(localizedPattern, locale).format(Date(time))
+    }
+
+    fun getHourFormat(c: Context?): String {
+        return if ((DateFormat.is24HourFormat(c))) "HH:mm" else "hh:mm a"
+    }
+
+    fun getDisplayFormattedTimeSpanString(c: Context, locale: Locale, timestamp: Long): String {
+        // If the timestamp is invalid (ie. 0) then assume we're waiting on data and just use the 'Now' copy
+        return if (timestamp == 0L || isWithin(timestamp, 1, TimeUnit.MINUTES)) {
+            // TODO ACL: We need a string for "Now" because I can't seem to coax a localised version from DateUtils -
+            // TODO: although anyone seeing is this is more than welcome to try! - 2024-08-26
+            "Now"
```

Comment:
Adding a note as this needs to get addressed

Reply:
"Now" isn't used anymore - I need to adjust `getDisplayFormattedTimeSpanString` as per SES-360. Am on it.